core/src/main/scala/org/http4s/QueryParam.scala

```diff
@@ -40,6 +41,43 @@ object QueryParamKeyLike {
   }
 }
 
+trait QueryParamCodec[T] extends QueryParamEncoder[T] with QueryParamDecoder[T]
+object QueryParamCodec {
+
+  def apply[A](implicit instance: QueryParamCodec[A]): QueryParamCodec[A] = instance
+
+  def from[A](decodeA: QueryParamDecoder[A], encodeA: QueryParamEncoder[A]): QueryParamCodec[A] =
+    new QueryParamCodec[A] {
+      override def encode(value: A): QueryParameterValue = encodeA.encode(value)
+      override def decode(value: QueryParameterValue): ValidatedNel[ParseFailure, A] =
+        decodeA.decode(value)
+    }
+
+  private def instantQueryParamDecoder(formatter: DateTimeFormatter): QueryParamDecoder[Instant] =
```

Comment:
Why would this and `instanctQueryParamDecoder` not go on the companions of their respective typeclasses?  The `instantQueryParamCodec` to join them makes sense, as does being explicit (because there's no default format).

Reply:
Good point, Ross. I'll address that.

Reply:
I initially thought you suggested moving `instantQueryParamDecoder` and `instantQueryParamEncoder` to `QueryParamDecoder` and `QueryParamEncoder` respectively. However, I already took that approach.

Could you please say more on your recommended change, @rossabaker?

Reply:
Aren't they in `QueryParamCodec` still?  Maybe I need to check this out and not squint at a GitHub diff.

Reply:
My mistake, Ross. You're right, `instantQueryParamDecoder` and `instantQueryParamEncoder` are defined in `object QueryParamCodec`. I'll move them to their respective companions. Sorry about that.

Reply:
Done in the commits added at https://github.com/http4s/http4s/pull/2724#commits-pushed-06b6a81.